### ISSUE TITLE
D3 v4.x support, no globals, AMD support

### ISSUE THREAD

**Issue by vlascik** [cimi/d3-flame-graphs#64]
Hello, this is a really nice graph, do you have a plan to rewrite this for D3 v4.x API (layout/nodes etc.), without using `d3` global, and supporting AMD, e.g. the way d3-tip does it? 
https://github.com/Caged/d3-tip/blob/master/index.js
```javascript
;(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module with d3 as a dependency.
    define([
      'd3-collection',
      'd3-selection'
    ], factory)
  } else if (typeof module === 'object' && module.exports) {
    /* eslint-disable global-require */
    // CommonJS
    var d3Collection = require('d3-collection'),
        d3Selection = require('d3-selection')
    module.exports = factory(d3Collection, d3Selection)
    /* eslint-enable global-require */
  } else {
    // Browser global.
    var d3 = root.d3
    // eslint-disable-next-line no-param-reassign
    root.d3.tip = factory(d3, d3)
  }
}(this, function(d3Collection, d3Selection) {
```
Thank you.

**Comment by cimi** [comment]
Thanks!

No plans to rewrite the plugin in its current form for D3 v4 - but happy to take in contributions if you want to add support!

I'm currently using this plugin in a webpack/ES6/babel/React project and integration in its current form is not great. As a next iteration I am planning to design it as a React component using D3 v4 under the hood - that will be a separate project.

**Comment by cimi** [comment]
I've pushed your PR to the `d3-v4` branch and released version `0.4.2-alpha` on npm. 

I decided not to release under a future version number (e.g. `v4.0.0-alpha`) because that would have caused problems in case I need to do any backwards incompatible change on the current form and need to bump up the major version. The reason it's `0.4.__2__-alpha` and not `0.4.0-alpha` is because I took out the UMD wrapper in the `.babelrc` and released, then I figured out what it was for and put it back. 😄 

Let me know if it works for you and if I can help with anything else.
